From de4983c069fa9975d2fdd67a22372f838c42a7e8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Fri, 19 Oct 2018 08:02:12 +0200
Subject: [PATCH] anaconda: make sure the latest version is placed as xen.efi
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

os.listdir returns files in filesystem order, not sorted.

QubesOS/qubes-issues#2990

Signed-off-by: Frédéric Pierret (fepitre) <frederic.epitre@orange.fr>
---
 pyanaconda/bootloader.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pyanaconda/bootloader.py b/pyanaconda/bootloader.py
index f3627e7dc..b3e76c933 100644
--- a/pyanaconda/bootloader.py
+++ b/pyanaconda/bootloader.py
@@ -1943,10 +1943,10 @@ class XenEFI(EFIGRUB):
         if os.path.exists(xen_efi_target):
             os.remove(xen_efi_target)
 
-        xen_efi = [x for x in os.listdir(util.getSysroot() + self.config_dir) if
-                   x.startswith('xen-') and x.endswith('.efi')][0]
+        xen_efi = [x for x in sorted(os.listdir(util.getSysroot() + self.config_dir)) if
+                   x.startswith('xen-') and x.endswith('.efi')][-1]
         shutil.copy("{}/{}".format(util.getSysroot() + self.config_dir, xen_efi),
-                    "{}/{}".format(util.getSysroot() + self.config_dir, "xen.efi"))
+                    xen_efi_target)
         rc = self.efibootmgr("-c", "-w", "-L", productName,
                              "-d", boot_disk.path, "-p", boot_part_num,
                              "-l",
-- 
2.19.2

